roles/insert_kernel_module/tasks/main.yml

```diff
@@ -1,39 +1,44 @@
 ---
-# This role loads one kernel_module specified in "kernel_module"
+# This role loads one kernel module specified in "insert_kernel_module_data"
 
-- fail:
+- name: "Fail on incorrect input : {{ insert_kernel_module_data.name }}"
+  ansible.builtin.fail:
     msg: >-
-      To run this role, you need to pass kernel_module object with attribute
+      To run this role, you need to pass insert_kernel_module_data object with attribute
       "name" defined.
-  when: kernel_module.name is not defined
+  when: insert_kernel_module_data.name is not defined
 
 # Detect what type of kernel module it is - builtin, loadable, or missing
-- import_tasks: detect_module.yml
+- name: "Import tasks to detect module type : {{ insert_kernel_module_data.name }}"
+  ansible.builtin.import_tasks: detect_module.yml
 
-# Sometimes it might be preffered to use the off-tree module instead of the module 
-# that is already shipped with the kernel. When a force_build flag is set for 
-# a module a build_module.yml will be called instead of loading it. 
+# Sometimes it might be preffered to use the off-tree module instead of the module
+# that is already shipped with the kernel. When a force_build flag is set for
+# a module a build_module.yml will be called instead of loading it.
 # NOTE(r-krcek): This needs testing, I am not sure if it works as-is!!
-- name: "{{ kernel_module.name }} : Fail if the user is trying to rebuild builtin module"
-  fail:
+- name: "Fail if the user is trying to rebuild builtin module : {{ insert_kernel_module_data.name }}"
+  ansible.builtin.fail:
     msg: >-
       Detected module_type is BUILTIN, but the force_build parameter was set to True.
       You cannot force build of a builtin module! Please check you kernel module configuration.
   when:
     - module_type == "builtin"
-    - kernel_module.force_build | default(False)
+    - insert_kernel_module_data.force_build | default(False)
 
-- name: "{{ kernel_module.name }} : Build module"
-  include_tasks: build_module.yml
-  when:
-    - kernel_module.force_build | default(False) or module_type == "missing"
+- name: Build and load the module
+  when: insert_kernel_module_data | check_cpu_constraints | bool
```

Comment:
The logic should be:
- no kernel module is inserted if the cpu constraints are not met. 
- thus, this check (check_cpu_constraints) should be done right after checking role parameters and if false, don't do any action inside this role.

Reply:
Thank you for the feedback. Should be addressed in 33036e3.